public/app/plugins/datasource/cloud-monitoring/components/QueryEditor.tsx

```diff
@@ -30,7 +25,7 @@ export class QueryEditor extends PureComponent<Props> {
       this.props.query.metricQuery = metricQuery;
     }
 
-    if (!this.props.query.hasOwnProperty('queryType')) {
+    if (this.props.query.queryType !== QueryType.METRICS || QueryType.SLO) {
```

Comment:
@andresmgot This change fixes [this issue](https://github.com/grafana/grafana/issues/44431#issuecomment-1200959230) which has been around for awhile apparently 🤷‍♂️

Reply:
got it, probably you didn't see for a while because it depends if you load the QueryEditor coming from a data source that does not use `queryType`. For example, for me that Azure Monitor is the default data source, `queryType` was `azure monitor` and since it was not empty it didn't enter this condition.

In any case it seems that this condition is not correct (if the query type is `slo` you are still changing it). Probably you meant:

```ts
    if (this.props.query.queryType !== QueryType.METRICS && this.props.query.queryType !== QueryType.SLO) {
```

or easier to read:

```ts
    if (![QueryType.METRICS, QueryType.SLO].includes(this.props.query.queryType)) {
```

Reply:
Ah yes, exactly 🏆!